hypothesis-python/docs/reference/api.rst

```diff
@@ -0,0 +1,1038 @@
+
+Hypothesis has an opaque binary representation that it uses for all examples it
+generates. This representation is not intended to be stable across versions or
+with respect to changes in the test, but can be used to to reproduce failures
+with the ``@reproduce_failure`` decorator.
+
+.. autofunction:: hypothesis.reproduce_failure
+
+The intent is that you should never write this decorator by hand, but it is
+instead provided by Hypothesis.
+When a test fails with a falsifying example, Hypothesis may print out a
+suggestion to use ``@reproduce_failure`` on the test to recreate the problem
+as follows:
+
+.. code-block:: pycon
+
+    >>> from hypothesis import settings, given, PrintSettings
+    >>> import hypothesis.strategies as st
+    >>> @given(st.floats())
+    ... @settings(print_blob=True)
+    ... def test(f):
+    ...     assert f == f
+    ...
+    >>> try:
+    ...     test()
+    ... except AssertionError:
+    ...     pass
+    ...
+    Falsifying example: test(f=nan)
+
+    You can reproduce this example by temporarily adding @reproduce_failure(..., b'AAAA//AAAAAAAAEA') as a decorator on your test case
+
+Adding the suggested decorator to the test should reproduce the failure (as
+long as everything else is the same - changing the versions of Python or
+anything else involved, might of course affect the behaviour of the test! Note
+that changing the version of Hypothesis will result in a different error -
+each ``@reproduce_failure`` invocation is specific to a Hypothesis version).
+
+By default these messages are not printed.
+If you want to see these you can set the :attr:`~hypothesis.settings.print_blob` setting to ``True``.
+
+
+.. _hypothesis-django:
+
+Django
+------
+
+.. seealso::
+
+    See the :ref:`Django strategies reference <django-strategies>` for documentation on strategies in the ``hypothesis.extra.django`` module.
+
+Hypothesis offers a number of features specific for Django testing, available
+in the ``hypothesis[django]`` :doc:`extra </extras>`.  This is tested
+against each supported series with mainstream or extended support -
+if you're still getting security patches, you can test with Hypothesis.
+
+.. autoclass:: hypothesis.extra.django.TestCase
+
+Using it is quite straightforward: All you need to do is subclass
+:class:`hypothesis.extra.django.TestCase` or
+:class:`hypothesis.extra.django.TransactionTestCase` or
+:class:`~hypothesis.extra.django.LiveServerTestCase` or
+:class:`~hypothesis.extra.django.StaticLiveServerTestCase`
+and you can use :func:`@given <hypothesis.given>` as normal,
+and the transactions will be per example
+rather than per test function as they would be if you used :func:`@given <hypothesis.given>` with a normal
+django test suite (this is important because your test function will be called
+multiple times and you don't want them to interfere with each other). Test cases
+on these classes that do not use
+:func:`@given <hypothesis.given>` will be run as normal.
```

Comment:
"as normal for <xref Django TestCase cls> or <ditto TransactionTestCase>."

Reply:
I'm not following what you're suggesting here sorry!

Reply:
Oh wow, markdown rendering issue!  Added backticks to clarify.

Reply:
wow, our django links go to a documentation site that I didn't even know existed!! https://django.readthedocs.io/en/stable/index.html#. I've updated this to the actual django docs at eg https://docs.djangoproject.com/en/5.1/topics/testing/tools/#django.test.TransactionTestCase